eth/backend.go

```diff
@@ -99,6 +105,31 @@ func (b *backend) SendTransaction(ctx context.Context, tx *types.Transaction) er
 	return nil
 }
 
+func (b *backend) SuggestGasPrice(ctx context.Context) (*big.Int, error) {
+	gp, err := b.Client.SuggestGasPrice(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	if b.maxGasPrice != nil && gp.Cmp(b.maxGasPrice) >= 0 {
```

Comment:
Minor: Should the conditional be > if we're maxGasPrice is the maximum allowable value (which is implied by the error message)?